docs/design/mesos-style.md

```diff
@@ -0,0 +1,162 @@
+for assigning the instances to machines.
+
+The flow of a request in Borg is:
+
+1. Client submits a collection object to the Borgmaster API endpoint
+1. Admission control, quota, applying defaults, etc. run on the collection
+1. If the collection is admitted, it is persisted, and the collection state machine creates the underlying instances
+1. The scheduler assigns a hostname to the instance, and tells the Borglet to start the instance's container(s)
+1. Borglet starts the container(s)
+1. The instance state machine manages the instances and the collection state machine manages the collection during their lifetimes
+
+Out-of-the-box Kubernetes has *workload-specific* abstractions (ReplicaSet, Job, DaemonSet, etc.) and corresponding controllers,
+and in the future may have [workload-specific schedulers](../../docs/proposals/multiple-schedulers.md),
+e.g. different schedulers for long-running services vs. short-running batch. But these abstractions, controllers, and
+schedulers are not *application-specific*.
+
+The usual request flow in Kubernetes is very similar, namely
+
+1. Client submits a collection object (e.g. ReplicaSet, Job, ...) to the API server
+1. Admission control, quota, applying defaults, etc. run on the collection
+1. If the collection is admitted, it is persisted, and the corresponding collection controller creates the underlying pods
+1. Admission control, quota, applying defaults, etc. runs on each pod; if there are multiple schedulers, one of the admission
+controllers will write the scheduler name as an annotation based on a policy
+1. If a pod is admitted, it is persisted
+1. The appropriate scheduler assigns a nodeName to the instance, which triggers the Kubelet to start the pod's container(s)
+1. Kubelet starts the container(s)
+1. The controller corresponding to the collection manages the pod and the collection during their lifetime
+
+Because Borg-style abstractions are not application-specific, it is common to end up with "two-level scheduling" when
+using Borg. For example, the MapReduce master might create a Borg Job with a certain number of instances with a particular resource shape,
+and then schedule the individual units of map and reduce work onto those instances itself.
+
+## What is a Mesos-style architecture?
+
+Mesos is fundamentally designed to support multiple application-specific "frameworks." A framework is
+composed of a "framework scheduler" and a "framework executor." We will abbreviate "framework scheduler"
+as "framework" since "scheduler" means something very different in Kubernetes (something that just
+assigns pods to nodes).
+
+Unlike Borg and Kubernetes, where there is a single logical endpoint that receives all API requests (the Borgmaster and API server,
+respectively), in Mesos every framework is a separate API endpoint. Mesos does not have any standard set of
+collection abstractions, controllers/state machines, or schedulers; the logic for all of these things is contained
```

Comment:
Added a mention at the end of the "What is a Mesos-style architecture" section along with a pointer to http://mesos.apache.org/documentation/latest/scheduler-http-api/